File: Algorithm/Programmers/level2_stack_주식가격.py

#case 1.
def solution(prices):
    answer = [0] * len(prices)

    for idx in range(len(prices)):
        cnt = -1
        for n in prices[idx:] :
            cnt+=1
            if prices[idx] > n : break
        answer[idx] = cnt

    return answer

File: Algorithm/Programmers/java/level2_stack_주식가격.java
/*
프로그래머스 level2 stack?

prices	return
[1, 2, 3, 2, 3]	[4, 3, 1, 1, 0]

https://school.programmers.co.kr/learn/courses/30/lessons/42584
*/

import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Arrays.fill(answer, 0);
        
        for(int i=0; i < prices.length; i++){
            for(int j=i+1; j < prices.length; j++){
                answer[i] += 1;
                if(prices[i] > prices[j]) break;
            }
        }
        
        return answer;
    }
}
